### Importing required packages

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

### Importing the dataset

In [2]:
MB_Train_data = pd.read_csv("train.csv")
MB_Test_data = pd.read_csv("test.csv")

### Performing EDA and Preprocessing on the given dataset

In [3]:
MB_Train_data.shape,MB_Test_data.shape

((4209, 378), (4209, 377))

In [4]:
MB_Train_data.size,MB_Test_data.size

(1591002, 1586793)

In [5]:
MB_Train_data.describe()

ID            y          X10     X11          X12  \
count  4209.000000  4209.000000  4209.000000  4209.0  4209.000000   
mean   4205.960798   100.669318     0.013305     0.0     0.075077   
std    2437.608688    12.679381     0.114590     0.0     0.263547   
min       0.000000    72.110000     0.000000     0.0     0.000000   
25%    2095.000000    90.820000     0.000000     0.0     0.000000   
50%    4220.000000    99.150000     0.000000     0.0     0.000000   
75%    6314.000000   109.010000     0.000000     0.0     0.000000   
max    8417.000000   265.320000     1.000000     0.0     1.000000   

               X13          X14          X15          X16          X17  ...  \
count  4209.000000  4209.000000  4209.000000  4209.000000  4209.000000  ...   
mean      0.057971     0.428130     0.000475     0.002613     0.007603  ...   
std       0.233716     0.494867     0.021796     0.051061     0.086872  ...   
min       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
25%       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
50%       0.000000     0.000000     0.000000     0.000000     0.000000  ...   
75%       0.000000     1.000000     0.000000     0.000000     0.000000  ...   
max       1.000000     1.000000     1.000000     1.000000     1.000000  ...   

              X375         X376         X377         X378         X379  \
count  4209.000000  4209.000000  4209.000000  4209.000000  4209.000000   
mean      0.318841     0.057258     0.314802     0.020670     0.009503   
std       0.466082     0.232363     0.464492     0.142294     0.097033   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       1.000000     0.000000     1.000000     0.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

              X380         X382         X383         X384         X385  
count  4209.000000  4209.000000  4209.000000  4209.000000  4209.000000  
mean      0.008078     0.007603     0.001663     0.000475     0.001426  
std       0.089524     0.086872     0.040752     0.021796     0.037734  
min       0.000000     0.000000     0.000000     0.000000     0.000000  
25%       0.000000     0.000000     0.000000     0.000000     0.000000  
50%       0.000000     0.000000     0.000000     0.000000     0.000000  
75%       0.000000     0.000000     0.000000     0.000000     0.000000  
max       1.000000     1.000000     1.000000     1.000000     1.000000  

[8 rows x 370 columns]

In [6]:
MB_Train_data.dtypes

ID        int64
y       float64
X0       object
X1       object
X2       object
         ...   
X380      int64
X382      int64
X383      int64
X384      int64
X385      int64
Length: 378, dtype: object

In [7]:
MB_Train_data.columns

Index(['ID', 'y', 'X0', 'X1', 'X2', 'X3', 'X4', 'X5', 'X6', 'X8',
       ...
       'X375', 'X376', 'X377', 'X378', 'X379', 'X380', 'X382', 'X383', 'X384',
       'X385'],
      dtype='object', length=378)

In [8]:
MB_Train_data = pd.DataFrame(MB_Train_data)

**Finding the unique values in the train datatset**

In [9]:
Column_values = MB_Train_data.values.ravel()
Unique_values = pd.unique(Column_values)
print(Unique_values)

[0 130.81 'k' ... 8415 87.48 8417]


**Finding the unique values in the test datatset**

In [10]:
Column_values2 = MB_Test_data.values.ravel()
Unique_values2 = pd.unique(Column_values2)
print(Unique_values2)

[1 'az' 'v' ... 8413 8414 8416]


**Droping the column named 'ID' from the train set**

In [11]:
MB_Train_data = MB_Train_data.drop(['ID'], axis = 1)
MB_Train_data

y  X0 X1  X2 X3 X4  X5 X6 X8  X10  ...  X375  X376  X377  X378  \
0     130.81   k  v  at  a  d   u  j  o    0  ...     0     0     1     0   
1      88.53   k  t  av  e  d   y  l  o    0  ...     1     0     0     0   
2      76.26  az  w   n  c  d   x  j  x    0  ...     0     0     0     0   
3      80.62  az  t   n  f  d   x  l  e    0  ...     0     0     0     0   
4      78.02  az  v   n  f  d   h  d  n    0  ...     0     0     0     0   
...      ...  .. ..  .. .. ..  .. .. ..  ...  ...   ...   ...   ...   ...   
4204  107.39  ak  s  as  c  d  aa  d  q    0  ...     1     0     0     0   
4205  108.77   j  o   t  d  d  aa  h  h    0  ...     0     1     0     0   
4206  109.22  ak  v   r  a  d  aa  g  e    0  ...     0     0     1     0   
4207   87.48  al  r   e  f  d  aa  l  u    0  ...     0     0     0     0   
4208  110.85   z  r  ae  c  d  aa  g  w    0  ...     1     0     0     0   

      X379  X380  X382  X383  X384  X385  
0        0     0     0     0     0     0  
1        0     0     0     0     0     0  
2        0     0     1     0     0     0  
3        0     0     0     0     0     0  
4        0     0     0     0     0     0  
...    ...   ...   ...   ...   ...   ...  
4204     0     0     0     0     0     0  
4205     0     0     0     0     0     0  
4206     0     0     0     0     0     0  
4207     0     0     0     0     0     0  
4208     0     0     0     0     0     0  

[4209 rows x 377 columns]

**Checking the null values in the train datatset**

In [12]:
MB_Train_data.isnull().sum()

y       0
X0      0
X1      0
X2      0
X3      0
       ..
X380    0
X382    0
X383    0
X384    0
X385    0
Length: 377, dtype: int64

**Checking the null values in the test datatset**

In [13]:
MB_Test_data.isnull().sum()

ID      0
X0      0
X1      0
X2      0
X3      0
       ..
X380    0
X382    0
X383    0
X384    0
X385    0
Length: 377, dtype: int64

**Using Labelencoder to convert the categorical data into numeric for the train set**

In [14]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
MB_Train_data['X0'] = le.fit_transform(MB_Train_data['X0'])
MB_Train_data['X1'] = le.fit_transform(MB_Train_data['X1'])
MB_Train_data['X2'] = le.fit_transform(MB_Train_data['X2'])
MB_Train_data['X3'] = le.fit_transform(MB_Train_data['X3'])
MB_Train_data['X4'] = le.fit_transform(MB_Train_data['X4'])
MB_Train_data['X5'] = le.fit_transform(MB_Train_data['X5'])
MB_Train_data['X6'] = le.fit_transform(MB_Train_data['X6'])
MB_Train_data['X8'] = le.fit_transform(MB_Train_data['X8'])

In [15]:
MB_Train_data

y  X0  X1  X2  X3  X4  X5  X6  X8  X10  ...  X375  X376  X377  \
0     130.81  32  23  17   0   3  24   9  14    0  ...     0     0     1   
1      88.53  32  21  19   4   3  28  11  14    0  ...     1     0     0   
2      76.26  20  24  34   2   3  27   9  23    0  ...     0     0     0   
3      80.62  20  21  34   5   3  27  11   4    0  ...     0     0     0   
4      78.02  20  23  34   5   3  12   3  13    0  ...     0     0     0   
...      ...  ..  ..  ..  ..  ..  ..  ..  ..  ...  ...   ...   ...   ...   
4204  107.39   8  20  16   2   3   0   3  16    0  ...     1     0     0   
4205  108.77  31  16  40   3   3   0   7   7    0  ...     0     1     0   
4206  109.22   8  23  38   0   3   0   6   4    0  ...     0     0     1   
4207   87.48   9  19  25   5   3   0  11  20    0  ...     0     0     0   
4208  110.85  46  19   3   2   3   0   6  22    0  ...     1     0     0   

      X378  X379  X380  X382  X383  X384  X385  
0        0     0     0     0     0     0     0  
1        0     0     0     0     0     0     0  
2        0     0     0     1     0     0     0  
3        0     0     0     0     0     0     0  
4        0     0     0     0     0     0     0  
...    ...   ...   ...   ...   ...   ...   ...  
4204     0     0     0     0     0     0     0  
4205     0     0     0     0     0     0     0  
4206     0     0     0     0     0     0     0  
4207     0     0     0     0     0     0     0  
4208     0     0     0     0     0     0     0  

[4209 rows x 377 columns]

**Droping the features having zero variance from the train set**

In [16]:
from sklearn.feature_selection import VarianceThreshold
Var_Threshold = VarianceThreshold(threshold=0)
Var_Threshold.fit(MB_Train_data)

VarianceThreshold(threshold=0)

In [17]:
Var_Threshold.get_support()

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True, False,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
       False,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True, False,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,

In [18]:
MB_Train_data.columns[Var_Threshold.get_support()]

Index(['y', 'X0', 'X1', 'X2', 'X3', 'X4', 'X5', 'X6', 'X8', 'X10',
       ...
       'X375', 'X376', 'X377', 'X378', 'X379', 'X380', 'X382', 'X383', 'X384',
       'X385'],
      dtype='object', length=365)

In [19]:
Zero_variance_col = [column for column in MB_Train_data.columns
                    if column not in MB_Train_data.columns[Var_Threshold.get_support()]]
print(len(Zero_variance_col))

12


In [20]:
for feature in Zero_variance_col:
    print(feature)

X11
X93
X107
X233
X235
X268
X289
X290
X293
X297
X330
X347


In [21]:
MB_Train_data.drop(Zero_variance_col, axis =1)

y  X0  X1  X2  X3  X4  X5  X6  X8  X10  ...  X375  X376  X377  \
0     130.81  32  23  17   0   3  24   9  14    0  ...     0     0     1   
1      88.53  32  21  19   4   3  28  11  14    0  ...     1     0     0   
2      76.26  20  24  34   2   3  27   9  23    0  ...     0     0     0   
3      80.62  20  21  34   5   3  27  11   4    0  ...     0     0     0   
4      78.02  20  23  34   5   3  12   3  13    0  ...     0     0     0   
...      ...  ..  ..  ..  ..  ..  ..  ..  ..  ...  ...   ...   ...   ...   
4204  107.39   8  20  16   2   3   0   3  16    0  ...     1     0     0   
4205  108.77  31  16  40   3   3   0   7   7    0  ...     0     1     0   
4206  109.22   8  23  38   0   3   0   6   4    0  ...     0     0     1   
4207   87.48   9  19  25   5   3   0  11  20    0  ...     0     0     0   
4208  110.85  46  19   3   2   3   0   6  22    0  ...     1     0     0   

      X378  X379  X380  X382  X383  X384  X385  
0        0     0     0     0     0     0     0  
1        0     0     0     0     0     0     0  
2        0     0     0     1     0     0     0  
3        0     0     0     0     0     0     0  
4        0     0     0     0     0     0     0  
...    ...   ...   ...   ...   ...   ...   ...  
4204     0     0     0     0     0     0     0  
4205     0     0     0     0     0     0     0  
4206     0     0     0     0     0     0     0  
4207     0     0     0     0     0     0     0  
4208     0     0     0     0     0     0     0  

[4209 rows x 365 columns]

**Droping the column named 'ID' from the test set**

In [22]:
MB_Test_data.drop(['ID'], axis = 1, inplace = True)

**Using Labelencoder to convert the categorical data into numeric for the test set**

In [23]:
from sklearn.preprocessing import LabelEncoder
le = LabelEncoder()
MB_Test_data['X0'] = le.fit_transform(MB_Test_data['X0'])
MB_Test_data['X1'] = le.fit_transform(MB_Test_data['X1'])
MB_Test_data['X2'] = le.fit_transform(MB_Test_data['X2'])
MB_Test_data['X3'] = le.fit_transform(MB_Test_data['X3'])
MB_Test_data['X4'] = le.fit_transform(MB_Test_data['X4'])
MB_Test_data['X5'] = le.fit_transform(MB_Test_data['X5'])
MB_Test_data['X6'] = le.fit_transform(MB_Test_data['X6'])
MB_Test_data['X8'] = le.fit_transform(MB_Test_data['X8'])

**Droping the features having zero variance from the test set**

In [24]:
from sklearn.feature_selection import VarianceThreshold
Var_Threshold2 = VarianceThreshold(threshold=0)
Var_Threshold2.fit(MB_Test_data)
Var_Threshold2.get_support()

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,

In [25]:
Zero_variance_col2 = [column for column in MB_Test_data.columns
                    if column not in MB_Test_data.columns[Var_Threshold2.get_support()]]
print(len(Zero_variance_col2))

5


In [26]:
MB_Test_data.drop(Zero_variance_col2, axis =1)

X0  X1  X2  X3  X4  X5  X6  X8  X10  X11  ...  X375  X376  X377  X378  \
0     21  23  34   5   3  26   0  22    0    0  ...     0     0     0     1   
1     42   3   8   0   3   9   6  24    0    0  ...     0     0     1     0   
2     21  23  17   5   3   0   9   9    0    0  ...     0     0     0     1   
3     21  13  34   5   3  31  11  13    0    0  ...     0     0     0     1   
4     45  20  17   2   3  30   8  12    0    0  ...     1     0     0     0   
...   ..  ..  ..  ..  ..  ..  ..  ..  ...  ...  ...   ...   ...   ...   ...   
4204   6   9  17   5   3   1   9   4    0    0  ...     0     0     0     0   
4205  42   1   8   3   3   1   9  24    0    0  ...     0     1     0     0   
4206  47  23  17   5   3   1   3  22    0    0  ...     0     0     0     0   
4207   7  23  17   0   3   1   2  16    0    0  ...     0     0     1     0   
4208  42   1   8   2   3   1   6  17    0    0  ...     1     0     0     0   

      X379  X380  X382  X383  X384  X385  
0        0     0     0     0     0     0  
1        0     0     0     0     0     0  
2        0     0     0     0     0     0  
3        0     0     0     0     0     0  
4        0     0     0     0     0     0  
...    ...   ...   ...   ...   ...   ...  
4204     0     0     0     0     0     0  
4205     0     0     0     0     0     0  
4206     0     0     0     0     0     0  
4207     0     0     0     0     0     0  
4208     0     0     0     0     0     0  

[4209 rows x 371 columns]

**Selecting the features and label from the train set and performing the split**

In [27]:
features = MB_Train_data.iloc[:,1:]
target= MB_Train_data.iloc[:,0]

In [28]:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(features, target, train_size=0.80, random_state = 4)

from sklearn.preprocessing import MinMaxScaler
scaling = MinMaxScaler().fit(X_train)
X_train_scaled = scaling.transform(X_train)
test_scaled = scaling.transform(MB_Test_data)

In [29]:
print(X_train.shape)
print(y_train.shape)
print(X_test.shape)
print(y_test.shape)

(3367, 376)
(3367,)
(842, 376)
(842,)


### Performing linear regression

In [30]:
from sklearn.linear_model import LinearRegression
lireg = LinearRegression()
lireg.fit(X_train,y_train)

LinearRegression()

In [31]:
y_pred = lireg.predict(X_test)
y_pred

array([9.19848633e+01, 9.34497070e+01, 1.10721191e+02, 9.56000977e+01,
       1.06713867e+02, 1.11549805e+02, 1.11716797e+02, 9.21025391e+01,
       9.49516602e+01, 9.62851562e+01, 9.57060547e+01, 1.17890137e+02,
       9.55913086e+01, 9.66889648e+01, 1.06465332e+02, 1.05773438e+02,
       9.75791016e+01, 9.27871094e+01, 1.04367676e+02, 1.05144043e+02,
       1.11470215e+02, 1.09887695e+02, 9.31967773e+01, 9.52104492e+01,
       9.16118164e+01, 9.46079102e+01, 9.30703125e+01, 9.38657227e+01,
       9.35332031e+01, 1.08580566e+02, 1.11842285e+02, 1.09410645e+02,
       8.71000977e+01, 9.50893555e+01, 7.74716797e+01, 1.11971191e+02,
       1.07784180e+02, 9.46206055e+01, 1.10869629e+02, 9.39819336e+01,
       1.11739746e+02, 1.04825684e+02, 9.33813477e+01, 9.53735352e+01,
       9.88212891e+01, 1.10405762e+02, 1.12986816e+02, 1.04784668e+02,
       9.30024414e+01, 9.42001953e+01, 8.94204102e+01, 9.62626953e+01,
       1.08240723e+02, 1.09164551e+02, 1.12312988e+02, 1.02771484e+02,
      

In [32]:
from sklearn.metrics import mean_squared_error
mean_squared_error(y_pred, y_test)

1.6910044773544357e+21

Above high value of MSE (1.69) suggests the biasedness or high variance or both in the given dataset.
Therefore, the model is not best fit for the given dataset.

To remove the biasedness in the dataset performing the Principal Component analysis. PCA will help to reduce the dimensionality of the given dataset

### Performing PCA

In [33]:
from sklearn.decomposition import PCA
pca = PCA(n_components = 'mle')
pca.fit(X_train)

PCA(n_components='mle')

In [34]:
pca.explained_variance_ratio_

array([3.81270883e-01, 2.14797517e-01, 1.32790472e-01, 1.18981873e-01,
       9.21027087e-02, 1.59702484e-02, 7.51224810e-03, 4.32566641e-03,
       2.94233662e-03, 2.41569033e-03, 2.36225481e-03, 2.07776071e-03,
       1.67484430e-03, 1.49616806e-03, 1.29875525e-03, 1.02977728e-03,
       9.52285227e-04, 8.90574218e-04, 7.99046329e-04, 7.27765175e-04,
       6.46244986e-04, 6.19201915e-04, 5.81315875e-04, 5.13638978e-04,
       4.96940089e-04, 4.34152731e-04, 4.05301443e-04, 3.80823107e-04,
       3.78072989e-04, 3.51024532e-04, 3.27872632e-04, 3.19533644e-04,
       2.99674893e-04, 2.83288386e-04, 2.69766191e-04, 2.44592498e-04,
       2.23415036e-04, 2.18367572e-04, 2.09092249e-04, 1.94361862e-04,
       1.88187849e-04, 1.80322404e-04, 1.76332015e-04, 1.73014496e-04,
       1.70087396e-04, 1.52415111e-04, 1.47780523e-04, 1.44702137e-04,
       1.42088498e-04, 1.39884285e-04, 1.34418476e-04, 1.28411949e-04,
       1.26629027e-04, 1.21300894e-04, 1.15979506e-04, 1.10000630e-04,
      

In [35]:
X_train_transformed = pca.transform(X_train)
X_test_transformed = pca.transform(X_test)

In [36]:
X_train_transformed.shape

(3367, 255)

As shown above the PCA has reduced the features from 376 to 255.

### Performing the linear regression on the data derived from the PCA

In [37]:
Lireg2 = LinearRegression()
Lireg2.fit(X_train_transformed, y_train)

LinearRegression()

In [38]:
y_pred2 = Lireg2.predict(X_test_transformed)
y_pred2

array([ 91.9880601 ,  93.45095176, 110.7172765 ,  95.60065046,
       106.71284444, 111.54849459, 111.71033837,  92.10591035,
        94.94336768,  96.27840657,  95.70616151, 117.88870969,
        95.59558187,  96.69363425, 106.46079618, 105.77608512,
        97.58282712,  92.8025246 , 104.3631028 , 105.13569259,
       111.46906824, 109.89078912,  93.19081601,  95.20339755,
        91.61335561,  94.61302909,  93.0777551 ,  93.87721749,
        93.5447986 , 108.58915336, 111.84816398, 109.40953005,
        87.10368098,  95.0966806 ,  77.46308038, 111.96702989,
       107.78450425,  94.63067239, 110.86608185,  93.98258733,
       111.73978511, 104.82137409,  93.37121361,  95.37225503,
        98.82221751, 110.4110682 , 112.9927015 , 104.77549491,
        93.00670246,  94.200985  ,  89.42066662,  96.27335936,
       108.25310625, 109.16553192, 112.321073  , 102.76123575,
        95.21076694,  93.84327552,  91.26447018,  92.73227928,
        94.24528119,  94.28757139, 111.24220353, 111.56

In [39]:
mean_squared_error(y_pred2, y_test)

74.0415830671749

Even after performing the PCA, the MSE value of the regression is still high 74.04 which makes the model less reliable.

To avoid the problem of biasedness and overfitting using the XGBoost algorithm

### Performing XGBoost algorithm

In [40]:
import xgboost as xgb
from sklearn.model_selection import GridSearchCV

In [41]:
grid_values = {'n_estimators': [74,75,76], 'learning_rate': [0.13,0.135,0.14], 'max_depth': [1,2,3]}
grid_xgb = GridSearchCV(xgb.XGBRegressor(), param_grid = grid_values, cv=10, scoring = 'r2')
grid_xgb.fit(X_train_transformed, y_train)

GridSearchCV(cv=10,
             estimator=XGBRegressor(base_score=None, booster=None,
                                    colsample_bylevel=None,
                                    colsample_bynode=None,
                                    colsample_bytree=None, gamma=None,
                                    gpu_id=None, importance_type='gain',
                                    interaction_constraints=None,
                                    learning_rate=None, max_delta_step=None,
                                    max_depth=None, min_child_weight=None,
                                    missing=nan, monotone_constraints=None,
                                    n_estimators=100, n_jobs=None,
                                    num_parallel_tree=None, random_state=None,
                                    reg_alpha=None, reg_lambda=None,
                                    scale_pos_weight=None, subsample=None,
                                    tree_method=None, validate_par

In [42]:
print('Mean score matrix: ', grid_xgb.cv_results_['mean_test_score'])
print('Grid best parameter (max. accuracy): ', grid_xgb.best_params_)
print('Grid best score (accuracy): ', grid_xgb.best_score_)

Mean score matrix:  [0.45764729 0.45814176 0.45913101 0.50777599 0.50801977 0.50839302
 0.51774343 0.51813768 0.51839086 0.46144273 0.46224533 0.46344747
 0.50973091 0.50963463 0.50961485 0.52625844 0.52614744 0.52578019
 0.46411203 0.46503366 0.46574995 0.50883041 0.50891536 0.50932526
 0.52558922 0.52552283 0.52541194]
Grid best parameter (max. accuracy):  {'learning_rate': 0.135, 'max_depth': 3, 'n_estimators': 74}
Grid best score (accuracy):  0.5262584397969535


The given R2 value suggests the variance of the dependent variable explained by the independent variable. A relatively high R2 value here, makes the model best fit.

In [43]:
predict_xgb = grid_xgb.predict(X_test_transformed)

**Saving the final prediction in a CSV file**

In [44]:
Predictions = pd.DataFrame()
Predictions['y'] = pd.Series(predict_xgb)
Predictions.to_csv('Predictions.csv')
Predictions

y
0     92.495544
1     97.489349
2    115.287903
3     97.873123
4    102.426994
..          ...
837   95.435631
838   97.034935
839  112.611992
840   94.571243
841  108.469093

[842 rows x 1 columns]